tests/unit/test_charm.py

```diff
@@ -267,6 +267,52 @@ def test__on_loki_push_api_endpoint_departed(
         self.assertEqual(path, "/etc/agent/agent.yaml")
         self.assertTrue(yaml.safe_load(content)["loki"] == {})
 
+    def test__loki_config_no_endpoints(self):
+        consumer = self.harness.charm._loki_consumer
+        endpoints = PropertyMock(return_value="")
+        type(consumer).loki_endpoints = endpoints
```

Comment:
Would `patch.object` work here instead of direct monkeypatching?

Reply:
IMO there is zero reason to mock here at all. We have had this discussion after the daily yesterday. This situation can be modelled using the relation-related capabilities of `Harness`, and a test written with `Harness` is way more valuable than one using mocks, because mocks are brittle, while the relations test the actual charm's contract :-)